docs(readme): add info about enforcer, notes, purge & reports (#114)
rHomelab/LabBot-Cogs: README.md

`[p]notes`
`[p]warnings`

Notes can be used for long-term storage information of problematic users or for future reference.

> When using listing the notes of a user, the warnings will be prioritised as they will be deemed more important.

Reviewer: ```suggestion
When listing the notes of a user, the warnings will be prioritised as they will be deemed more important.
```

> It is possible to add notes to a user who has not joined or has left. Notes are stored against a user's ID and not necessarily requiring the user to be part of the server.

Reviewer: ```suggestion
It is possible to add notes to a user who has left or not yet joined. Notes are stored against a user's ID and do not require the user to be part of the server.
```

> Deleting a note does not actually delete the note. It simply marks it in the datastore as deleted. This can allow future recovery of notes if a bad party is attempting to wipe existing notes, but will not be visible in lists of normal use.

Reviewer: ```suggestion
Deleting a note does not actually delete the note. It simply marks it in the datastore as deleted and does not show it in normal use. This can allow future recovery of notes if a bad party is attempting to wipe existing notes.
```
